packages/ui/src/components/Form/expression/ExpressionEditor.tsx

```diff
@@ -25,79 +29,221 @@ export const ExpressionEditor: FunctionComponent<ExpressionEditorProps> = ({
+      <TextInputGroup isPlain>
+        <TextInputGroupMain
+          data-testid="typeahead-select-input"
+          value={inputValue}
+          onClick={onToggleClick}
+          onChange={onTextInputChange}
+          onKeyDown={onInputKeyDown}
+          id="typeahead-select-input"
+          autoComplete="off"
+          innerRef={textInputRef}
+          placeholder={SchemaService.DROPDOWN_PLACEHOLDER}
+          {...(activeItem && { 'aria-activedescendant': activeItem })}
+          role="combobox"
+          isExpanded={isOpen}
+          aria-controls="select-typeahead-listbox"
+        />
+
+        <TextInputGroupUtilities>
+          {!!inputValue && (
+            <Button
+              data-testid="clear-input-value"
+              variant="plain"
+              onClick={() => {
+                setSelected('');
+                setInputValue('');
+                setFilterValue('');
+                onChangeExpressionModel('', {});
+                textInputRef?.current?.focus();
+              }}
+              aria-label="Clear input value"
+            >
+              <TimesIcon aria-hidden />
+            </Button>
+          )}
+        </TextInputGroupUtilities>
+      </TextInputGroup>
+    </MenuToggle>
   );
 
   return (
     languageCatalogMap && (
       <>
-        <Dropdown
-          id="expression-select"
-          data-testid="expression-dropdown"
+        <Select
+          id="typeahead-select"
           isOpen={isOpen}
-          selected={selected !== '' ? selected : undefined}
+          selected={selected}
           onSelect={onSelect}
-          onOpenChange={setIsOpen}
+          onOpenChange={() => {
+            setIsOpen(false);
+          }}
           toggle={toggle}
-          isScrollable={true}
         >
-          <DropdownList data-testid="expression-dropdownlist">
-            {Object.values(languageCatalogMap).map((lang) => {
-              return (
-                <DropdownItem
-                  data-testid={`expression-dropdownitem-${lang.model.name}`}
-                  key={lang.model.title}
-                  value={lang.model.name}
-                  description={lang.model.description}
-                >
-                  {lang.model.title}
-                </DropdownItem>
-              );
-            })}
-          </DropdownList>
-        </Dropdown>
+          <SelectList id="select-typeahead-listbox">
+            {selectOptions.map((option, index) => (
+              <SelectOption
+                key={option.value as string}
+                description={option.description}
+                isFocused={focusedItemIndex === index}
+                className={option.className}
+                data-testid={`expression-dropdownitem-${option.value}`}
+                onClick={() => setSelected(option.children as string)}
+                id={`select-typeahead-${option.value.replace(' ', '-')}`}
+                {...option}
+                ref={null}
```

Comment:
Could be possible this is not needed? :thinking:

Reply:
You're right @lordrip , removed.